src/Microsoft.DotNet.Build.Tasks/PackageFiles/scripts/xplatagent/cleanupagents.py

```diff
@@ -0,0 +1,127 @@
+import glob, datetime, os, re, sys
+
+class agentFolder:
+  def __init__(self, sourceFolderJson, agent_builddirectory, lastRunOn):
+    self.sourceFolderJson = sourceFolderJson
+    self.agent_builddirectory = agent_builddirectory
+    self.lastRunOn = lastRunOn
+    dateMatch = re.match("(\d+)/(\d+)/(\d+)", self.lastRunOn)
+    self.month = int(dateMatch.group(1))
+    self.day = int(dateMatch.group(2))
+    self.year = int(dateMatch.group(3))
+    
+  def isValid(self):
+    validDateMatch = re.match('\d+/\d+/\d+ \d+:\d+:\d+ -\d+:\d+', self.lastRunOn)
+    validDirectoryMatch = re.match('^\S+', self.agent_builddirectory)
+    if validDateMatch is None:
+      print 'invalid date'
+      return False
+    if validDirectoryMatch is None:
+      print self.agent_builddirectory
+      print 'invalid dir'
+      return False
+    return True
+
+# main method 
+def main():
+  try:
+    agentDirectory = sys.argv[1]
+    retainDays = sys.argv[2]
+  except:
+    usage()
+  # validate we were pointed to an xplat agent directory root
+  if not os.path.isdir(agentDirectory + "/_work"):
+    print 'agent directory \'' + agentDirectory + '/_work\' does not exist'
+    usage()
+  print 'Agent directory:', agentDirectory
+
+  # find all of the sourcefolder.json files - https://github.com/Microsoft/vso-agent/blob/master/docs/workfolder.md
+  sourceFolderJsonPaths = agentDirectory + '/_work/SourceRootMapping/*/*/SourceFolder.json'
+  sourceFolderJsons = glob.glob(sourceFolderJsonPaths)
+
+  agents = []
+  # determine the datetime for which we will delete agent working directories older than
+  nowDateTime = datetime.datetime.now()
+  nowtimedelta = datetime.timedelta(days=float(retainDays))
+  cleanupDateTime = nowDateTime - nowtimedelta
+
+  # create out agent objects
+  for sourceFolderJson in sourceFolderJsons:
+    agent = CreateAgentObject(sourceFolderJson)
+    if agent.isValid() == True:
+      agents.append(agent)
+    else:
+      print "Warning: unable to parse source folder '" + sourceFolderJson + "'"
+
+  # remove agents older than specified # of days
+  for agent in agents:
+    agentDateTime = datetime.datetime(agent.year, agent.month, agent.day)
+    if agentDateTime < cleanupDateTime:
+      print "cleanup agent " + agent.agent_builddirectory + " which ran on " + agent.lastRunOn
+      cleanupDirectory = agentDirectory + "/_work/" + agent.agent_builddirectory
+      print "cleanup directory: " + cleanupDirectory
+      RemoveAgentDirectory(cleanupDirectory, agent.sourceFolderJson)
+
+# removes the agent working directory (cleanupDirectory) and its sourceFolderJson file
+def RemoveAgentDirectory(cleanupDirectory, sourceFolderJson):
```

Comment:
I think you can simplify this by using shutil.rmtree, see https://docs.python.org/2/library/shutil.html#shutil.rmtree